plugins/test/a_module.py

```diff
@@ -38,20 +38,26 @@
   type: boolean
 """
 
+import typing as t
+from collections.abc import Callable
+
 from ansible.plugins.loader import action_loader, module_loader
+from ansible.errors import AnsibleFilterError
 
 try:
     from ansible.errors import AnsiblePluginRemovedError
 except ImportError:
     AnsiblePluginRemovedError = Exception  # type: ignore
 
 
-def a_module(term):
+def a_module(term: t.Any) -> bool:
     """
     Example:
       - 'community.general.ufw' is community.general.a_module
       - 'community.general.does_not_exist' is not community.general.a_module
     """
+    if not isinstance(term, str):
```

Comment:
Why not
```suggestion
def a_module(term: str) -> bool:
    """
    Example:
      - 'community.general.ufw' is community.general.a_module
      - 'community.general.does_not_exist' is not community.general.a_module
    """
    if not isinstance(term, str):
```
?

Reply:
Since Ansible does not ensure tha tthe input is actually a string, `t.Any` is the only correct input type for test and filter inputs. The plugin itself has to ensure that it is of the correct type.

(Also if you put `term: str`, then mypy will say that the code inside the `if` is dead code since it can never be reached.)